src/components/views/rooms/MessageComposerInput.js

```diff
@@ -56,29 +63,54 @@ function mdownToHtml(mdown) {
 /*
  * The textInput part of the MessageComposer
  */
-module.exports = React.createClass({
-    displayName: 'MessageComposerInput',
+export default class MessageComposerInput extends React.Component {
+    constructor(props, context) {
+        super(props, context);
+        this.onAction = this.onAction.bind(this);
+        this.onInputClick = this.onInputClick.bind(this);
+        this.handleReturn = this.handleReturn.bind(this);
+        this.handleKeyCommand = this.handleKeyCommand.bind(this);
+        this.onChange = this.onChange.bind(this);
+
+        this.state = {
+            isRichtextEnabled: false, // TODO enable by default when RTE is mature enough
+            editorState: null
+        };
 
-    statics: {
-        // the height we limit the composer to
-        MAX_HEIGHT: 100,
-    },
+        // bit of a hack, but we need to do this here since createEditorState needs isRichtextEnabled
+        this.state.editorState = this.createEditorState();
 
-    propTypes: {
-        tabComplete: React.PropTypes.any,
+        this.client = MatrixClientPeg.get();
+    }
 
-        // a callback which is called when the height of the composer is
-        // changed due to a change in content.
-        onResize: React.PropTypes.func,
+    static getKeyBinding(e: SyntheticKeyboardEvent): string {
+        // C-m => Toggles between rich text and markdown modes
+        if(e.keyCode == 77 && KeyBindingUtil.isCtrlKeyCommand(e)) {
```

Comment:
We'd been taking these out as constants which makes them a little easier to read.